zenodo/modules/deposit/utils.py

```diff
@@ -245,3 +247,33 @@ def suggest_language(q, limit=5):
     if lut:
         langs = ([lut, ] + [l for l in langs if l != lut])[:limit]
     return langs
+
+
+def is_user_verified(days=7):
+    """
+    Permission function that evaluates if the user can create a deposit.
+
+    This applies in addition to preexisting permissions.
+    """
+    from flask_security import current_user
+    if current_user.external_identifiers:
+        return True
+
+    error_message = (
+            'To create a deposit please verify your email.'
+            .format('/account/settings/linkedaccounts/'))
+    if not current_user.confirmed_at:
+        return (False, error_message)
+    error_message = (
+        'You are using a domain for your email address that generates '
```

Comment:
You have registered on Zenodo using an email address domain that have recently uploaded spam on Zenodo. Your account needs to have been verified for at least X days before you can publish. <LINK TEXT>.